snapd-glib/requests/snapd-json.c

```diff
@@ -176,13 +176,18 @@ _snapd_json_get_date_time (JsonObject *object, const gchar *name)
 }
 
 static void
-parse_error_response (JsonObject *root, GError **error)
+parse_error_response (JsonObject *root, JsonNode **error_value, GError **error)
 {
     JsonObject *result = _snapd_json_get_object (root, "result");
     gint64 status_code = _snapd_json_get_int (root, "status-code", 0);
     const gchar *kind = result != NULL ? _snapd_json_get_string (result, "kind", NULL) : NULL;
     const gchar *message = result != NULL ? _snapd_json_get_string (result, "message", NULL) : NULL;
 
+    if (error_value != NULL) {
+        *error_value = result != NULL ? json_object_get_member (result, "value") : NULL;
+        json_node_ref (*error_value);
```

Comment:
`json_node_ref` will hit an assertion if this in NULL (not documented, but I checked the json-glib source). So probably safer to do:
```c
JsonNode *node = NULL;
if (result != NULL)
    node = json_object_get_member (result, "value");
*error_value = node != NULL ? json_node_ref (node) : NULL;
```

Reply:
I've made the json_node_ref call conditional on *error_value being not-NULL.